test/server/config_validation/xds_fuzz_test.cc

```diff
@@ -0,0 +1,21 @@
+#include "common/protobuf/utility.h"
+
+#include "test/fuzz/fuzz_runner.h"
+#include "test/server/config_validation/xds_fuzz.h"
+#include "test/server/config_validation/xds_fuzz.pb.validate.h"
+
+namespace Envoy {
+
+DEFINE_PROTO_FUZZER(const test::server::config_validation::XdsTestCase& input) {
+  RELEASE_ASSERT(!TestEnvironment::getIpVersionsForTest().empty(), "");
+  try {
+    TestUtility::validate(input);
+  } catch (const ProtoValidationException& e) {
+    ENVOY_LOG_MISC(debug, "ProtoValidationException: {}", e.what());
+    return;
+  }
+  XdsFuzzTest test(input, envoy::config::core::v3::ApiVersion::V2);
```

Comment:
Hm. Fuzzing V3 probably makes sense to start

Reply:
Sure.  This will require a small refactor as `Config::TypeUrl::get().Cluster` used in `compareDiscoveryRequest` returns the v2 type url.  Should I do that here or open a new PR for that?

Reply:
Interesting, it does look like ads_integration_test is primarily tested with v2. @htuch is it intentional?